### ISSUE TITLE
Instant frame rate drop 

### ISSUE THREAD

**Issue by teaNoPot** [Genymobile/scrcpy#3221]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: Windows 10
 - scrcpy version: 1.23
 - installation method: Windows release
 - device model: Xiaomi Redmi Note 8 Pro
 - Android version: 10

**Describe the bug**
I execute the command `scrcpy.exe` normally through the cmd, there is a bit of a stagger while it loads the server. Upon the start of the screen capture, I get an immediate drop in frame rate :

```

C:\Users\xxx\xxx\scrcpy-win64-v1.23>scrcpy.exe --print-fps
scrcpy 1.23 <https://github.com/Genymobile/scrcpy>
C:\Users\xxx\xxx\scrcpy-win64-v1.23\scrcpy-server:... file pushed, 0 skipped. 51.9 MB/s (41123 bytes in 0.001s)
[server] INFO: Device: Xiaomi Redmi Note 8 Pro (Android 10)
INFO: Renderer: direct3d
INFO: Initial texture: 1080x2336
INFO: FPS counter started
INFO: 62 fps (+5 frames skipped)
INFO: 0 fps
INFO: 16 fps
INFO: 0 fps
INFO: 118 fps (+20 frames skipped)
INFO: 4 fps
INFO: 0 fps
INFO: 1 fps
INFO: 0 fps
```

The drops seem to be coherent with what I get from the visual feedback. I tried using different arguments : 
- limiting frame rate.
- limiting the size.
- changing the renderer to opengl.
- lowering the bitrate.

But to no avail. I will keep looking for the problem in hopes I can find its source, but in the meantime any help would be appreciated :)

Have a good day !

**Comment by rom1v** [comment]
A frame is produced only when the device screen changes. So if you play a video at 25fps, you'll get 25 fps. If you display a static screen, then 0fps is expected.

**Comment by teaNoPot** [comment]
> A frame is produced only when the device screen changes. So if you play a video at 25fps, you'll get 25 fps. If you display a static screen, then 0fps is expected.

I see your point, however the drop of frames happen as I'm using the phone (I'm scrolling through the actual phone, tapping on apps and so on...). The window capture does not show ultimately these events on screen thus the 0fps. When I launch an app for example, I start getting something like this: 

```
INFO: 13 fps (+2 frames skipped)
INFO: 0 fps
INFO: 14 fps (+1 frames skipped)
INFO: 0 fps
INFO: 12 fps (+3 frames skipped)
INFO: 0 fps
INFO: 13 fps (+2 frames skipped)
INFO: 0 fps
INFO: 13 fps (+2 frames skipped)
INFO: 0 fps
INFO: 13 fps (+2 frames skipped)
```

On the window capturing my screen, I can clearly see this problem.

**Comment by rom1v** [comment]
If you record, is the recorded file laggy?

```
scrcpy --record file.mp4
```

**Comment by teaNoPot** [comment]
Affirmative, the first few ms the frames look somewhat normal then it drops into a freezing state (0fps)

**Comment by rom1v** [comment]
OK, then the problem is the capture on the device.

Does it  work with screenrecord:

```
adb shell screenrecord /sdcard/file.mp4
adb pull /sdcard/file.mp4
```

Could you try with another device?

**Comment by teaNoPot** [comment]
Yes it does work normally with screenrecord, I tried using another device and have encountered the same freezing problem (I used : Xiaomi M2101K6G on Android 12). I do not have other devices to test with for now.

**Comment by rom1v** [comment]
> Affirmative, the first few ms the frames look somewhat normal then it drops into a freezing state (0fps)

Could you share the file recorded by scrcpy?

**Comment by teaNoPot** [comment]
Sure :

https://user-images.githubusercontent.com/15705349/165532045-99407132-03f8-4b30-b7c3-3f81d4d16527.mp4
